packages/lexical-playground/src/plugins/TableHoverActionsPlugin/index.tsx

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import {useLexicalComposerContext} from '@lexical/react/LexicalComposerContext';
+import {
+  $getTableColumnIndexFromTableCellNode,
+  $getTableRowIndexFromTableCellNode,
+  $insertTableColumn__EXPERIMENTAL,
+  $insertTableRow__EXPERIMENTAL,
+  $isTableCellNode,
+  $isTableNode,
+  TableCellNode,
+  TableNode,
+  TableRowNode,
+} from '@lexical/table';
+import {$findMatchingParent} from '@lexical/utils';
+import {$getNearestNodeFromDOMNode} from 'lexical';
+import {useEffect, useRef, useState} from 'react';
+import * as React from 'react';
+import {createPortal} from 'react-dom';
+
+import {useDebounce} from '../CodeActionMenuPlugin/utils';
+
+const BUTTON_WIDTH_PX = 20;
+
+function TableHoverActionsContainer({
+  anchorElem,
+}: {
+  anchorElem: HTMLElement;
+}): JSX.Element {
+  const [editor] = useLexicalComposerContext();
+  const [isShownRow, setShownRow] = useState<boolean>(false);
+  const [isShownColumn, setShownColumn] = useState<boolean>(false);
+  const [shouldListenMouseMove, setShouldListenMouseMove] =
+    useState<boolean>(false);
+  const [position, setPosition] = useState({});
+  const codeSetRef = useRef<Set<string>>(new Set());
+  const tableDOMNodeRef = useRef<HTMLElement | null>(null);
+
+  const debouncedOnMouseMove = useDebounce(
+    (event: MouseEvent) => {
+      const {isOutside, tableDOMNode} = getMouseInfo(event);
+
+      if (isOutside) {
+        setShownRow(false);
+        setShownColumn(false);
+        return;
+      }
+
+      if (!tableDOMNode) {
+        return;
+      }
+
+      tableDOMNodeRef.current = tableDOMNode;
+
+      let hoveredRowNode: TableCellNode | null = null;
+      let hoveredColumnNode: TableCellNode | null = null;
+      let tableDOMElement: HTMLElement | null = null;
+
+      editor.update(() => {
+        const maybeTableCell = $getNearestNodeFromDOMNode(tableDOMNode);
+
+        if ($isTableCellNode(maybeTableCell)) {
+          const table = $findMatchingParent(maybeTableCell, (node) =>
+            $isTableNode(node),
+          );
+          if (!table) {
+            return;
+          }
+
+          tableDOMElement = editor.getElementByKey(table?.getKey());
+
+          if (tableDOMElement) {
+            const rowCount = (table as TableNode).getChildrenSize();
```

Comment:
... ⭐️ some other cases below as well (mostly a nit tho)
```suggestion
            const rowCount = table.getChildrenSize();
```